tools/ARIAtools/extractProduct.py

```diff
@@ -591,28 +628,108 @@ def export_products(full_product_dict, bbox_file, prods_TOTbbox, layers, rankedR
+        if not os.path.exists(workdir):
+            os.mkdir(workdir)
+
+        # Iterate through all IFGs
+        for i in enumerate(product_dict[0]):
+            wt_outname = os.path.abspath(os.path.join(workdir, start_key
+                                         + product_dict[1][i[0]][0]))
+            ##Update progress bar
+            prog_bar.update(i[0]+1,suffix=product_dict[1][i[0]][0])
+
+            # make VRTs for wet and then dry components
+            hgt_field = prep_metadatalayers(wt_outname, [i[1]][0], dem)
+            hy_outname = os.path.abspath(os.path.join(workdir, end_key
+                                         + product_dict[1][i[0]][0]))
+            tropo_prefix = '/science/grids/corrections/external/troposphere/'
+            hydro_comp = [i.replace(tropo_prefix+start_key, \
+                          tropo_prefix+end_key) \
+                          for i in [i[1]][0]]
+            hgt_field = prep_metadatalayers(hy_outname, hydro_comp, dem)
+
+
+            # initiate raster
+            outname = os.path.abspath(os.path.join(workdir,
+                                      product_dict[1][i[0]][0]))
+            print(outname)
+            # building the VRT
+            gdal.BuildVRT(outname + '.vrt', wt_outname + '.vrt')
+            gdal.Open(outname + '.vrt').SetMetadataItem(
+                      hgt_field, gdal.Open(wt_outname + '.vrt'
+                                            ).GetMetadataItem(hgt_field))
+            gdal.Warp(outname, outname + '.vrt',
+                      options=gdal.WarpOptions(format=outputFormat))
+            # Update VRT
+            gdal.Translate(outname + '.vrt', outname,
+                           options=gdal.TranslateOptions(format="VRT"))
+            # Update VRT with new raster
+            update_file = gdal.Open(outname, gdal.GA_Update)
+            heightsMeta = np.array(gdal.Open(wt_outname + '.vrt' \
+                          ).GetMetadataItem(hgt_field)[1:-1].split(','), \
+                          dtype='float32')
+            wet_arr = gdal.Open(wt_outname + '.vrt').ReadAsArray()
+            hy_arr = gdal.Open(hy_outname + '.vrt').ReadAsArray()
+            for i in range(len(heightsMeta)):
+                mean_arr = (wet_arr[i] + hy_arr[i]) / 2
+                update_file.GetRasterBand(i+1).WriteArray(mean_arr)
+            del update_file
+            # Interpolate/intersect with DEM before cropping
+            finalize_metadata(outname, bounds, dem_bounds, \
+                              prods_TOTbbox, dem, lat, lon, hgt_field, \
+                              mask, outputFormat, verbose=verbose)
+
+            #If necessary, resample raster
+            if multilooking is not None and key!='unwrappedPhase' \
+                 and key!='connectedComponents':
+                resampleRaster(outname, multilooking, bounds, prods_TOTbbox,
+                               rankedResampling, outputFormat=outputFormat,
```

Comment:
Does one even get here? 
- the if statement is only for troposphere.

Could we not do only work at the end for tropo-total?
you should be able to sum the hydro and wet cubes without need to manipulate and then stitch them?

Reply:
Yeah, had copied from below with haste. I've removed this 'if' condition, but we need to be able to resample the array if the user so specifies.

And yes, the idea is to first stitch the cubes for hydro and wet cubes into temp files before summing. I will confirm this behaves as intended with tests spanning two frames.